pages/index.js

```diff
@@ -1,121 +1,173 @@
 import Head from 'next/head'
-import Header from './nav'
+import { Navbar } from '../components/Navbar'
 import Footer from './footer'
 import products from '../products.json';
 import { initiateCheckout } from '../lib/payments';
+import { useState } from 'react';
 import Stripe from '@stripe/stripe-js';
 
 
+const defaultCart = {
+  products: {}
+}
+
+
 export default function Home() {
 
-  console.log('NEXT_PUBLIC_STRIPE_API_KEY', process.env.NEXT_PUBLIC_STRIPE_API_KEY)
+  const [cart, updateCart] = useState(defaultCart);
+
+  const cartItems = Object.keys(cart.products).map(id => {
+    const product = products.find(({ id }) => `${id}`);
```

Comment:
```suggestion
    const product = products.flat()[id];
```

Amb el metodo `Array.prototype.flat` podem convertir un array bidimensional (Array de arrays) en un array unidimensional. Amb aço conseguim un array que que tots els productes y ahi ja podem usar el id per a trobar el producte.


https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat